### ISSUE TITLE
Support instance derivation for types parametric on a barbie-type

### ISSUE THREAD

**Issue by jcpetruzza** [jcpetruzza/barbies#5]
The following currently doesn't work:

```haskell
data B b (f :: * -> *)  = B (b f)
  deriving (Generic)

instance FunctorB b => FunctorB (B b)
```

Of course, one can write the instances manually...
```haskell
instance FunctorB b => FunctorB (B b) where
    bmap f (B bf) = B (bmap f bf)
```

**Comment by jcpetruzza** [comment]
`FunctorB` seems to be working now on `master` for this case, but not `ConstraintsB`

**Comment by jcpetruzza** [comment]
Fixed in v1.0.0.0
